### PR TITLE
Added Bin-Sort algorithm for C# from Issue#432

### PR DESCRIPTION
**Pull request by zjjiang2**
**Fixes [#432](https://github.com/FOSS-UCSC/FOSSALGO/issues/432)**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

**Thank you for the opportunity, please let me know if there is any changes that needs to be made.**

### CODE REVIEW

File: algorithms/ar-binsrt/csharp/bin-sort.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+			return;
+		}
+    }
+
+    public static void Main(){
+        float[] arr = new float[] {0.102F, 0.212F, 0.442F, 0.661F, 0.232F, 0.157F, 0.341F};
+        DoBinSort(arr, (arr.Length));
+        foreach(float element in arr)
+        {
+            Console.Write(element + " ");
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/csharp/bin-sort.cs:47]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6341744)

File: algorithms/ar-binsrt/csharp/bin-sort.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            index = 0;
+            for (int i = 0; i < n; i++) {
+                for (int j = 0; j < bins[i].Count; j++) {
+                    arr[index++] = bins[i][j];
+                }
+            }
+		}
+		else{
+            //End the function if the array is empty.
+			return;
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/csharp/bin-sort.cs:38]: Codacy found an issue: [Remove this redundant jump.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6341744)

File: algorithms/ar-binsrt/csharp/bin-sort.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    public static void Main(){
+        float[] arr = new float[] {0.102F, 0.212F, 0.442F, 0.661F, 0.232F, 0.157F, 0.341F};
+        DoBinSort(arr, (arr.Length));
+        foreach(float element in arr)
+        {
+            Console.Write(element + " ");
+        }
+        Console.WriteLine("");
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/csharp/bin-sort.cs:49]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6341744)

File: algorithms/ar-binsrt/csharp/bin-sort.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            }
+		}
+		else{
+            //End the function if the array is empty.
+			return;
+		}
+    }
+
+    public static void Main(){
+        float[] arr = new float[] {0.102F, 0.212F, 0.442F, 0.661F, 0.232F, 0.157F, 0.341F};
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/csharp/bin-sort.cs:43]: Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6341744)

File: algorithms/ar-binsrt/csharp/bin-sort.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public static void Main(){
+        float[] arr = new [] {0.102F, 0.212F, 0.442F, 0.661F, 0.232F, 0.157F, 0.341F};
+        DoBinSort(arr, (arr.Length));
+        foreach(float element in arr)
+        {
+            //Console.Write(element + " ");
+        }
+        //Console.WriteLine("");
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/csharp/bin-sort.cs:45]: Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6341744)

File: algorithms/ar-binsrt/csharp/bin-sort.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            }
+		}
+    }
+
+    public static void Main(){
+        float[] arr = new [] {0.102F, 0.212F, 0.442F, 0.661F, 0.232F, 0.157F, 0.341F};
+        DoBinSort(arr, (arr.Length));
+        foreach(float element in arr)
+        {
+            //Console.Write(element + " ");
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/csharp/bin-sort.cs:43]: Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6341744)

File: algorithms/ar-binsrt/csharp/bin-sort.cs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+                }
+            }
+		}
+    }
+
+    public static void Main(){
+        float[] arr = new [] {0.102F, 0.212F, 0.442F, 0.661F, 0.232F, 0.157F, 0.341F};
+        DoBinSort(arr, (arr.Length));
+        foreach(float element in arr)
+        {
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/csharp/bin-sort.cs:42]: Codacy found an issue: [Either remove or fill this block of code.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6341744)

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@zjjiang2 would you be able to add this contribution to the chart of algorithms in the main readme file. Don't change anything else.

**Review by RavinduSachintha — APPROVED** [review]

**Comment by zjjiang2** [comment]
README.md has been updated.

**Comment by RavinduSachintha** [comment]
@all-contributors please add @zjjiang2 for code and doc

**Comment by RavinduSachintha** [comment]
@zjjiang2 thank you for your contribution. Keep it up 👍
Make sure to star this repo 😁

**Pull request merged**